### PR TITLE
feat: Lift and implement methods from WalletProvider to NetworkInfoProvider

### PR DESCRIPTION
**Pull request by IvayloAndonov**
# Context

As part of the refactor to remove `WalletProvider`, `ledgerTip`, `currentWalletProtocolParameters`, and `genesisParameters` should be moved to `NetworkInfoProvider`, and then the `WalletProvider` removed completely. The `DbSyncProvider` needs to also be extended to support these new methods.

# Proposed Solution
N/A

# Important Changes Introduced

**New feature section:** (bae10f3ee3bf8a53d488e6d306f96b1a64473edd)
- extend `NetworkInfoHttpService` with new endpoints: `/tip`, `/genesis-params` and `/protocol-params`
- implement `DbSynNetworkInfoProvider` methods lifted from `WalletProvider` interface
- create db queries and mappers
- extend `openApi.json` schema
- add http service tests

**Refactoring section:** (c549ed450e39fc577a0b4d87459261d037870cb6)
- core: extend NetworkInfoProvider, remove WalletProvider interface
- cardano-services-client: extend networkInfoHttpProvider with methods
- blockfrost: lifted methods from wallet to networkInfo provider, tests
- wallet: remove TrackedWalletProvider, extend TrackedNetworkInfoProvide
- wallet: update and fix all e2e, load and integration tests
- web-extension: update e2e tests

### CODE REVIEW

**Review by rhyslbw — CHANGES_REQUESTED** [review]
As nice as it is to shorten the URLs, please make the following changes to ensure alignment with the provider interface, which is the pattern implemented up to this point:

- `/genesis-params` -> `/genesis-parameters`
- `/protocol-params` -> `/current-wallet-protocol-parameters`

File: packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/mappers.ts [hunk 1]
```diff
@@ -46,16 +53,64 @@ export const toNetworkInfo = ({
...
     const genesisFilePath = require(path.resolve(cardanoNodeConfigPath)).ShelleyGenesisFile;
     const genesis = JSONbig({ useNativeBigInt: true }).parse(
       fs.readFileSync(path.resolve(path.dirname(cardanoNodeConfigPath), genesisFilePath), 'utf-8')
     );
+
     return {
+      activeSlotsCoefficient: genesis.activeSlotsCoeff,
+      epochLength: genesis.epochLength,
+      maxKesEvolutions: genesis.maxKESEvolutions,
       maxLovelaceSupply: genesis.maxLovelaceSupply,
```
Comment (rhyslbw) [inline packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/mappers.ts:106]: This actually highlights an opportunity to improve our `NetworkInfoProvider` interface, by exposing `maxLovelaceSupply` and `networkId` under `genesisParameters`, which is already part of the `CompactGenesis` type. That would leave `NetworkInfo.network.timeSettings`, which we can consider lifting as a new top-level provider method, and then remove the `NetworkInfo` type by hoisting the remaining properties too i.e.

**Before**
``` ts
export type NetworkInfo = {
  network: {
    id: Cardano.NetworkId;
    magic: Cardano.NetworkMagic;
    timeSettings: TimeSettings[];
  };
  lovelaceSupply: AssetSupply;
  stake: StakeSummary;
};

export interface NetworkInfoProvider {
  currentWalletProtocolParameters(): Promise<ProtocolParametersRequiredByWallet>;
  genesisParameters(): Promise<Cardano.CompactGenesis>;
  ledgerTip(): Promise<Cardano.Tip>;
  networkInfo(): Promise<NetworkInfo>;
}
```

**After**
``` ts
export interface NetworkInfoProvider {
  currentWalletProtocolParameters(): Promise<ProtocolParametersRequiredByWallet>;
  genesisParameters(): Promise<Cardano.CompactGenesis>;
  lovelaceSupply(): Promise<AssetSupply>;
  stake(): Promise<StakeSummary>;
  timeSettings(): Promise<TimeSettings[]>;
}
```

**Review by mkazlauskas — CHANGES_REQUESTED** [review]
Looking good, just need to remove wallet provider from all `.env.example` files and resolve lint errors

File: packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/mappers.ts (hunk 1 above)
Comment (mkazlauskas) [reply packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/mappers.ts:106]: Agree, should probably be out of scope of this PR though

Comment (rhyslbw) [reply packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/mappers.ts:106]: I think we should at least do the genesis consolidation here, as right now it's a bit confusing on this aspect.

File: packages/cardano-services/src/NetworkInfo/openApi.json [hunk 2]
```diff
@@ -173,6 +254,137 @@
...
+            "type": "number",
+            "example": 49032415
+          },
+          "hash": {
+            "type": "string",
+            "example": "5caede44f4a5a775443095159cd42c8a64f35494086957ab3e04624015a6e13c"
+          }
+        }
+      },
+      "WalletProtocolParamsResponse": {
```
Comment (rhyslbw) [inline packages/cardano-services/src/NetworkInfo/openApi.json:280]:
```suggestion
      "CurrentWalletProtocolParametersResponse": {
```
Any exposed API should used the complete name

File: packages/cardano-services/src/NetworkInfo/openApi.json [hunk 3]
```diff
@@ -173,6 +254,137 @@
...
+              },
+              "major": {
+                "type": "number",
+                "example": 2
+              }
+            }
+          }
+        }
+      },
+      "GenesisParamsResponse": {
```
Comment (rhyslbw) [inline packages/cardano-services/src/NetworkInfo/openApi.json:346]:
```suggestion
      "GenesisParametersResponse": {
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]
When applying feedback this time, try using `fixup` commits, so you're able to `autosquash` before merge :sunglasses:

File: packages/cardano-services/src/NetworkInfo/openApi.json (hunk 2 above)
Comment (IvayloAndonov) [reply packages/cardano-services/src/NetworkInfo/openApi.json]: de39caa57407d90f0623275fc237cb8737040a13

File: packages/cardano-services/src/NetworkInfo/openApi.json (hunk 3 above)
Comment (IvayloAndonov) [reply packages/cardano-services/src/NetworkInfo/openApi.json]: de39caa57407d90f0623275fc237cb8737040a13

**Review by mkazlauskas — APPROVED** [review]

File: packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/mappers.ts (hunk 1 above)
Comment (rhyslbw) [reply packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/mappers.ts:106]: To be done in ADP-1911

**Review by mkazlauskas — DISMISSED** [review]

**Comment by rhyslbw** [comment]
Fixup of another misnamed URL path 5fec9230c1c8321945272546d7d36e12707c8d8c.

**Pull request merged**